### ISSUE TITLE
MetaValueProcessor and v3

### ISSUE THREAD

**Issue by scottk4016** [XiaoFaye/WooCommerce.NET#410]
I have been working on a project to initially load Customers, Products and Orders in from WooCommerce using v3 APIs to a local database and then maintain the database using WebHooks updates from WooCommerce.  It works well, except we have complex Order metadata from a WooCommerce plugin. We really need that additional metadata in the database. 

Using the instructions in the post https://github.com/XiaoFaye/WooCommerce.NET/wiki/Handle-different-types-of-Meta-Value-in-WC-Restful-API-V2 I have been able to create a test project where I have successfully used MetaValueProcessor to get access to the complex meta data. Thanks for the great post James :-)

Is it possible to use MetaValueProcessor with the v3 APIs?  If not, I am going to have to change my project to use v2 APIs and then make some other minor changes to map the v2 classes to the database schema.  Since I am already using Json.Net to deserialise the Webhooks I think I am with handle the metadata there.  Just looking for some advice on whether I am on the right track, before I dive in to making changes to use the v2 APIs.  

Thanks
Scott

**Comment by XiaoFaye** [comment]
Yes, it will work on v3 APIs as well.

**Comment by scottk4016** [comment]
Thanks for the reply. I'm glad to hear that, but looks like I am doing something wrong. When I use a v2 using directive it works correctly and runs the MetaValueProcessor. However, when I change the using directive to v3 (and change the RestAPI url parameter to v3) the code runs but it never calls the MetaValueProcessor.  I set a breakpoint in ProcessMetaValue and it hits it when I use v2, but not when using v3.  Any suggestions?

Code posted below with RestAPI details obfuscated.

Thanks
Scott

Program.cs
```C#
using System;
using System.Collections.Generic;
using WooCommerceNET.WooCommerce.v3;
using Newtonsoft.Json.Linq;

namespace ConsoleApp1
{
    class Program
    {
        static void Main(string[] args)
        {
            new WcLoad().Run();
        }
    }

    class WcLoad : WCObject
    {
        private static readonly MyRestAPI rest = new MyRestAPI(
            "https://www.abcxyz.com.au/wp-json/wc/v3/", 
            "ck_012345789012345789012345789012345789", 
            "cs_012345789012345789012345789012345789", 
            false);

        public WcLoad() : base(rest)
        {
            WCObject.MetaValueProcessor = ProcessMetaValue;
        }

        public Order Run()
        {
            List<Order> list = new List<Order>();
            foreach (var order in Order.GetAll().Result)
                list.Add(order);

            return null;
        }

        private object ProcessMetaValue(string parentTypeName, object value)
        {
            if (value != null && value.GetType().FullName.StartsWith("Newtonsoft.Json"))
            {
                if (value.GetType().FullName.EndsWith("JArray"))
                {
                    JArray ja = (JArray)Convert.ChangeType(value, typeof(JArray));
                    if (ja.HasValues)
                    {
                        try
                        {
                            return ja.ToObject<List<Dictionary<string, string>>>();
                        }
                        catch { }

                        return value;
                    }
                    else
                        return null;
                }
                else if (value.GetType().FullName.EndsWith("JObject"))
                {
                    JObject jo = (JObject)Convert.ChangeType(value, typeof(JObject));

                    if (jo.HasValues)
                    {
                        return GetRaqMeta(jo);
                    }
                    else
                        return null;
                }
                else
                    return value;
            }
            else
                return value;
        }

        private Dictionary<string, string> GetRaqMeta(JObject jo)
        {
            var list = new Dictionary<string, string>();

            foreach (var child in jo)
            {
                var key = child.Key;
                var obj = child.Value;

                if (obj?.Type != JTokenType.Object)
                    continue;

                if (obj?.Value<string>("id") == null)
                    continue;

                var val = child.Value?.Value<string>("value");
                list.Add(key, val);
            }
            return list;
        }
    }
}
```

MyRestAPI.cs
```C#
using System;
using System.Net;
using WooCommerceNET;
using Newtonsoft.Json;

namespace ConsoleApp1
{
    public class MyRestAPI : RestAPI
    {
        public MyRestAPI(string url, string key, string secret, bool authorizedHeader = true,
            Func<string, string> jsonSerializeFilter = null,
            Func<string, string> jsonDeserializeFilter = null,
            Action<HttpWebRequest> requestFilter = null) : base(url, key, secret, authorizedHeader, jsonSerializeFilter, jsonDeserializeFilter, requestFilter)
        {
        }

        public override T DeserializeJSon<T>(string jsonString)
        {
            return JsonConvert.DeserializeObject<T>(jsonString);
        }

        public override string SerializeJSon<T>(T t)
        {
            return JsonConvert.SerializeObject(t);
        }
    }
}
```

**Comment by XiaoFaye** [comment]
I see what the problem now, some classes in V3 inherit from V2, but the trigger point is in the V3 MetaObject, I should fix this in the next version ASAP.

**Comment by scottk4016** [comment]
Great. Thanks for that.

**Comment by XiaoFaye** [comment]
Hi @scottk4016 , this issue is fixed in version 0.8.0.
